transcribe/scribe_modules/cpu_vulnerabilities.py

```diff
@@ -12,19 +14,13 @@ def __init__(self, input_dict=None, module_name=None, host_name=None,
                                        input_type=input_type,
                                        scribe_uuid=scribe_uuid)
         if input_dict:
-            self._assign_properties(self._parse(input_dict))
-
-
-    def __iter__(self):
-        for attr, value in self.__dict__.items():
-            yield attr, value
-
-    def _assign_properties(self, initial_data):
-        for key in initial_data:
-            setattr(self, key, initial_data[key])
+            self._input_dict = input_dict
+        else:
+            raise ValueError('Input dictionary is empty')
 
-    def _parse(self, input_parse):
-        output_dict = {}
+    def parse(self):
+        input_parse = self._input_dict
+        output_dict = copy.deepcopy(self._dict)
```

Comment:
Deep copy can be very, very expensive.  The only time you need a copy is if you are going to expose a python object that somebody is going to modify.  Can we avoid this by fixing the need to modify the objects exposed?

Reply:
We want to yield multiple objects per invocation, so i'm not sure there'll be a way to do that.

Reply:
nvm its not needed in a few modules.. i am going to remove that